storyruntime/processing/Lexicon.py

```
@@ -236,6 +236,34 @@ def unless_condition(logger, story, line):
             return line['exit']
         return line['enter']
 
+    @staticmethod
+    async def try_catch(logger, story, line):
+        from . import Stories
+        next_line = story.next_block(line)
+
+        if next_line is None:
+            raise StoryscriptError(
+                message=f'Try requires catch block',
+                story=story, line=line)
+
+        try:
+            await Stories.execute_block(logger, story, line)
+        except BaseException as e:
+            output = next_line['output'][0]
+            try:
+                story.context[output] = str(e)
```

Uhm at the team summit we decided to not expose the exception message (or other context) yet to the user, but only allow him to differentiate between exceptions (user errors, recoverable, e.g. 404), and errors (application error, not recoverable, e.g. 500).

More details: https://github.com/storyscript/storyscript/issues/954

---

I'd love to sync up on this and get a full breakdown. I'll review that issue, and be sure to ping you with any questions.

---

We're going to focus on implementing basic try catch as the docs currently outline. We can re-visit this in the future.

---

So that means we don't support access to the error object, right and this should be removed.
I'm aware that https://docs.storyscript.io/storyscript/advanced/#exception-handling mentions `as error`, but we _never_ defined how that object should look like.

---

If you take a look at this issue I just submitted, https://github.com/storyscript/storyscript/issues/1094

`as error` was working perfectly for me not too long ago, with the try catch implementation on the runtime of course. Trying to debug it, I realized I could simply define the variable globally to get the runtime to compile. When passing this to the context, I realized I was deleting it after the catch block executes. In the end, removing the global variable I defined. It's a bit of a weird situation, but the above is more or less a temporary fix to that problem.

I'm not sure if the compiler should throw an error since the variable was already defined, but I am experiencing an issue a bit intertwined in this.